src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -90,16 +95,7 @@ protected virtual bool CanUpdateProperty(ModelMetadata propertyMetadata)
                 return true;
             }
 
-            // 3. The model name is not prefixed and a value provider can directly provide a value for the model name.
```

Comment:
from commit comments:
> - change `MutableObjectModelBinder` to ignore exact match in value providers
 - had an incorrect assumption: don't want exact model name to match since
   this binder supports only complex objects
 - ignored `BinderModelName`, value provider filtering, et cetera